test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -688,6 +689,27 @@ public void RootCauseLocalization()
             }
         }
 
+        [Theory]
+        [InlineData(-1, 3)]
+        [InlineData(40, 3)]
+        [InlineData(20, -1)]
+        public void TestDetectSeasonality(int seasonalityWindowSize, int expectedPeriod)
+        {
+            // Create a detect seasonality input
+            var input = new List<double>() {18.004, 87.401, 87.411, 18.088, 18.017, 87.759, 33.996, 18.043, 87.853, 18.364, 18.004, 86.992, 87.555,
+                18.088, 18.029, 87.906, 87.471, 18.039, 18.099, 87.403, 18.030, 72.991, 87.804, 18.381, 18.016, 87.145, 87.771,
```

Comment:
Same comment as the one in the sample. It would be great if this code works with data generated from a periodic function and we could demonstrate that.

Reply:
changed to `sin(2 * pi + x)`